File: API_SC_01_08Abril/Api_Alvaro_Pietro/q2.py
#inacabada
def main():
    peso_atual = float(input('Seu peso atual: '))
    sexo = input('Você é homem ou mulher (H OU M)?: ').upper()
    quilos_perder = float(input('Quantos quilos você deseja perder?: '))
    dias_semana_treino = int(input('Quantos dias da semana você vai treinar?: '))
    tempo_por_dia = int(input('Quantos minutos você ira treinar por dia: '))
    tempo_transport = int(input('Quanto do seu tempo, em %, sera destinado para o TRANSPORT?: '))/100
    tempo_escada = 1 - tempo_transport

    peso_inicial = peso_atual - quilos_perder
    calorias_homem = 2400
    perder_1kg = 7000
    calorias_homem = 2000

    tempo_dia_transport = tempo_por_dia * tempo_transport
    tempo_transport_semana = tempo_dia_transport * dias_semana_treino

    tempo_dia_escada = tempo_por_dia * tempo_escada
    tempo_escada_semana = tempo_dia_escada * dias_semana_treino

    if sexo == 'H':
        while tempo_transport_semana > 0:
            tempo_transport_semana - 5
            calorias_homem = calorias_homem - 100

        while tempo_escada_semana > 0:
            tempo_escada_semana - 7
            calorias_homem = calorias_homem - 100
        

        if calorias_homem < 0:
            calorias_queimadas = calorias_homem / perder_1kg
            peso_atual = peso_atual - calorias_queimadas
            return peso_atual
        return peso_atual

    if peso_atual <= peso_inicial:
        semana = dias_semana_treino + dias_semana_treino
        return semana


        
    print(f'quantas semanas para o objetivo: {semana}')
    print(f'tempo de exercicio transport por dia:{tempo_dia_transport}')
    print(f'tempo de exercicio transport por dia: {tempo_dia_escada}')

main()

File: API_SC_01_08Abril/FelipeMadson/q2.js
import { question, questionInt } from 'readline-sync';

function calcularTempoPerdaPeso(pesoAtual, caloriasNecessarias, caloriasAtividade, diasPorSemana, proporcaoTransport) {
    const caloriasExcedentesPorSemana = (caloriasNecessarias - caloriasAtividade * diasPorSemana * (proporcaoTransport / 100));
    const semanasParaPerderPeso = caloriasExcedentesPorSemana / 7000; 

    return semanasParaPerderPeso;
}

function calcularTempoAtividadePorDia(semanasParaPerderPeso, diasPorSemana, proporcaoTransport) {
    const minutosPorSemana = semanasParaPerderPeso * diasPorSemana * 7 * 24; 
    const minutosTransportPorSemana = minutosPorSemana * (proporcaoTransport / 100);
    const minutosEscadaPorSemana = minutosPorSemana - minutosTransportPorSemana;

    const minutosTransportPorDia = minutosTransportPorSemana / diasPorSemana;
    const minutosEscadaPorDia = minutosEscadaPorSemana / diasPorSemana;

    return [minutosTransportPorDia, minutosEscadaPorDia];
}

function main() {
    console.log("Calculadora de Perda de Peso");

    const pesoAtual = questionInt("Qual é o seu peso atual? ");
    const genero = question("Você é homem ou mulher? (H/M): ").toLowerCase();
    const pesoPerder = questionInt("Quantos quilos você deseja perder? ");
    const diasPorSemana = questionInt("Quantos dias por semana você irá treinar? ");
    const tempoPorDia = questionInt("Quanto tempo (em minutos) você irá treinar por dia? ");
    const proporcaoTransport = questionInt("Qual proporção (%) de tempo você alocará para o Transport? ");

    let caloriasNecessarias;
    if (genero === 'h') {
        caloriasNecessarias = 2400;
    } else if (genero === 'm') {
        caloriasNecessarias = 2000;
    } else {
        console.log("Por favor, insira um gênero válido (H ou M).");
        return;
    }

    const caloriasAtividade = (genero === 'h') ? (tempoPorDia / 5) * 100 : (tempoPorDia / 6) * 100;

    const semanasParaPerderPeso = calcularTempoPerdaPeso(pesoAtual, caloriasNecessarias, caloriasAtividade, diasPorSemana, proporcaoTransport);
    const [minutosTransportPorDia, minutosEscadaPorDia] = calcularTempoAtividadePorDia(semanasParaPerderPeso, diasPorSemana, proporcaoTransport);

    console.log(`\nPara perder ${pesoPerder}kg, você precisará de aproximadamente ${semanasParaPerderPeso.toFixed(1)} semanas.`);
    console.log(`\nEm cada dia de treino:\n- ${minutosTransportPorDia.toFixed(0)} minutos de Transport`);
    console.log(`- ${minutosEscadaPorDia.toFixed(0)} minutos de Escada`);
}

main();
